test/martha/martha_v3.test.js

```diff
@@ -424,412 +382,332 @@ test.serial('martha_v3 returns an error when an invalid field is requested', asy
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'anvil';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
+    sinon.assert.callCount(getJsonFromApiStub, 2);
 });
 
 test.serial('martha_v3 parses a The AnVIL CIB URI response correctly', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(anvilDrsResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('anvil');
+    const drs = drsUrls(config.HOST_THE_ANVIL_STAGING, 'dg.ANV0%2F00008531-03d7-418c-b3d3-b7b22b5381a0');
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(anvilDrsResponse);
     const response = mockResponse();
+
     await marthaV3(
         mockRequest({ body: { 'url': 'drs://dg.ANV0:dg.ANV0/00008531-03d7-418c-b3d3-b7b22b5381a0' } }),
         response,
     );
-    t.is(response.statusCode, 200);
 
-    const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 2); // Bond was called to get SA key
-    t.deepEqual({ ...result }, anvilDrsMarthaResult(googleSAKeyObject, null));
-    t.is(
-        getJsonFromApiStub.getCall(0).args[0],
-        `https://${config.HOST_THE_ANVIL_STAGING}/ga4gh/drs/v1/objects/dg.ANV0%2F00008531-03d7-418c-b3d3-b7b22b5381a0`,
-    );
-    t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
+    t.is(response.statusCode, 200);
+    t.deepEqual(response.body, anvilDrsMarthaResult(googleSAKeyObject, null));
 
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'anvil';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
+    sinon.assert.callCount(getJsonFromApiStub, 2);
 });
 
 test.serial('martha_v3 parses Kids First response correctly', async (t) => {
-    const drsAccessUrlResponse = mockS3AccessUrl(kidsFirstDrsResponse.access_methods[0].access_url.url);
-    getJsonFromApiStub.onCall(0).resolves(kidsFirstDrsResponse);
-    getJsonFromApiStub.onCall(1).resolves(bondAccessTokenResponse);
-    getJsonFromApiStub.onCall(2).resolves(drsAccessUrlResponse);
+    const {
+        id: objectId, self_uri: drsUri,
+        access_methods: { 0: { access_id: accessId, access_url: { url: s3Url } } },
+    } = kidsFirstDrsResponse;
+    const drsAccessUrlResponse = mockS3AccessUrl(s3Url);
+    const bond = bondUrls('kids-first');
+    const drs = drsUrls(config.HOST_KIDS_FIRST_PROD, objectId, accessId);
```

Comment:
I think I did this in other places as well, but this is a case where I had to change the host from `STAGING` to `PROD`... I think in order to match the hosts encoded in the test data. I _think_ this is okay, but please take a minute to consider this while reviewing and reply here with your thoughts. Thanks.

Reply:
If this was like most non-TDR DRS hosts, their prod servers probably existed before test/staging servers. So I suspect that the test data was generated from a prod example. Now that staging exists maybe a new example can be generated for the unit tests.

Reply:
Ah, good point! Sounds like a good idea to me, now that (I'm pretty sure) there is a Kids First staging server.